src/RobotSimulator/Arenas/challengeSlice.ts

```diff
@@ -0,0 +1,57 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { RootState } from "../../state/store";
+
+export enum ChallengeStatus {
+  Pending,
+  Failure,
+  Success,
+}
+
+export type ChallengeInfo = {
+  status: ChallengeStatus;
+  id: string;
+};
+
+export const challengeSlice = createSlice({
+  initialState: {
+    challengeInfos: [] as ChallengeInfo[],
+  },
+  name: "challenge",
+  reducers: {
+    setChallengeStatus(
```

Comment:
The update logic in this reducer is probably a bit hard to read.  I need to think further how it could be simplified.  Maybe we could pass the current challenge state into the `ChallengeActionsImpl` constructor, and then the ChallengeActionsImpl object can report success/failure based on the events it receives and the current challenge state.  That way there would be less logic in here ...